### PR TITLE
Password optional

### PR DESCRIPTION
**Pull request by bryanapellanes-okta**
- Add method to check if password is required.
- Update sample app to check if password field should be rendered.
- Tested flow with `identifier first` deactivated and flow functioned properly (video not shown).

https://user-images.githubusercontent.com/63638027/158620416-2449329d-6ca2-4c1b-9d85-e8fc49330374.mp4

### CODE REVIEW

File: src/Okta.Idx.Sdk.UnitTests/IdxClientShould.cs [hunk 1]
```diff
@@ -15341,5 +15341,262 @@ await function.Should()
...
+    ""intent"": ""LOGIN"",
+    ""remediation"": {
+        ""type"": ""array"",
+        ""value"": [
+            {
+                ""rel"": [
+                    ""create-form""
+                ],
+                ""name"": ""identify"",
+                ""href"": ""https://vm-test-oie.trexcloud.com/idp/idx/identify"",
```
Comment (laura-rodriguez) [inline src/Okta.Idx.Sdk.UnitTests/IdxClientShould.cs:15366]: can you pls mask the test org in the mocks?

File: src/Okta.Idx.Sdk/IdxClient.cs [hunk 2]
```diff
@@ -1902,5 +1902,30 @@ public async Task<PollResponse> PollAuthenticatorPushStatusAsync(IIdxContext idx
...
+                throw new ArgumentNullException(nameof(idxContext));
+            }
+
+            var introspectResponse = await this.IntrospectAsync(idxContext, cancellationToken);
+
+            return new PasswordRequiredResponse
+            {
+                Context = idxContext,
+                IsPasswordRequired = IsRemediationRequireCredentials("identify", introspectResponse),
+                Response = introspectResponse,
```
Comment (laura-rodriguez) [inline src/Okta.Idx.Sdk/IdxClient.cs:1927]: is there any reason why we're returning the raw response? We usually don't include it as part of other responses (by design).

**Review by laura-rodriguez — COMMENTED** [review]
This looks good overall. I left a few comments to review before approval.

File: src/Okta.Idx.Sdk/IdxClient.cs (hunk 2 above)
Comment (bryanapellanes-okta) [reply src/Okta.Idx.Sdk/IdxClient.cs:1927]: I thought it would be useful for debugging if it became necessary.  I don't feel strongly about keeping it.

**Review by laura-rodriguez — APPROVED** [review]

**Pull request merged**